File: java/leetcode/src/main/java/com/leetcode/list/SwapNodes_in_Pairs_24.java
package com.leetcode.list;

public class SwapNodes_in_Pairs_24 {
    public ListNode swapPairs(ListNode head) {
        ListNode dummy = new ListNode();
        ListNode pre = dummy;

        ListNode cur = head;
        while (cur != null) {
            ListNode first = cur;
            ListNode second = cur.next;
            if (second != null) {
                cur = second.next;
                pre.next = second;
                second.next = first;
                first.next = null;
                pre = first;
            } else {
                pre.next = first;
                break;
            }
        }

        return dummy.next;
    }
}

File: js/algrithm/huahua130/list/Swap_Nodes _in_Pairs_24.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var swapPairs = function(head) {
    const dummy = new ListNode(-1, undefined)
    let pre = dummy
    let cur = head
    while (cur) {
        let first = cur
        let second = cur?.next
        if (second) {
            cur = second.next
            pre.next = second
            second.next = first
            first.next = null
            pre = first
        } else {
            pre.next = first
            break
        }
    }
    return dummy.next

};